dotCMS/src/main/java/com/dotmarketing/util/ImportExportUtil.java

```diff
@@ -1407,33 +1447,61 @@ private void doXMLFileImport(File f, PrintWriter out, ObjectFilter filter)throws
-                                HibernateUtil.save(obj);
-                            }
-                            else{
-                                Logger.warn(this.getClass(), "Can't import tree- no matching inodes: {parent=" + t.getParent() + ", child=" + t.getChild() +"}");
-
-                            }
-                        }
-                        else{
+                            final Tree t = (Tree) obj;
+                            LocalTransaction.wrap(() -> {
+
+                                DotConnect dc = new DotConnect();
+                                List<String> inodeList = new ArrayList<String>();
+                                dc.setSQL("select inode from inode where inode = ? or inode = ?");
+                                dc.addParam(t.getParent());
+                                dc.addParam(t.getChild());
+                                inodeList = dc.getResults();
+                                dc.setSQL("select id from identifier where id = ? or id = ?");
+                                dc.addParam(t.getParent());
+                                dc.addParam(t.getChild());
+                                inodeList.addAll(dc.getResults());
+                                if(inodeList.size() > 1){
+                                    HibernateUtil.save(obj);
+                                }
+                                else {
+                                    Logger.warn(this.getClass(), "Can't import tree- no matching inodes: {parent=" + t.getParent() + ", child=" + t.getChild() +"}");
+                                }
+                            });
+                        } else{
                             try {
+
+                                Logger.debug(this, "Saving the object: " +
+                                        obj.getClass() + ", with the values: " + obj);
+
+                                HibernateUtil.startTransaction();
                                 HibernateUtil.save(obj);
+                                HibernateUtil.commitTransaction();
                             } catch (DotHibernateException e) {
                                 Logger.error(this,e.getMessage(),e);
+                                try {
+
+                                    Logger.debug(this, "Error on trying to save: " +
+                                            e.getMessage()
+                                            + ", trying to Save the object again: " +
+                                            obj.getClass() + ", with the values: " + obj);
+
+                                    HibernateUtil.save(obj);
+                                    HibernateUtil.commitTransaction();
+                                }catch (Exception ex) {
+                                    Logger.debug(this, "Usually not a problem can be that duplicate data or many times a row of data that is created by the system and is trying to be imported again : " + ex.getMessage(), ex);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 225 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)